src/RegExpr.pas

```diff
@@ -1201,8 +1200,8 @@ constructor TRegExpr.Create;
  begin
   inherited;
   programm := nil;
-  fExpression := nil;
-  fInputString := nil;
+  fExpression := '';
```

Comment:
Are you sure this is safe to do?
I mean empty string ('') is not logically equivalent to not assigned (nil).
Of cause no sense to have empty expression but of cause you can empty input.
And even for expression - I just do not like this assumption 'nobody needs empty re'. What if..

Reply:
What is safe? setting empty str in Create? of course it's safe, and i even can delete this setting to empty str, because objects always have EMPTY str fields on creation

Reply:
My point we better distinguish between unassigned and assigned empty string.
Unassigned - error, empty string - no error.

Reply:
With pascal strings we cannot distinguish it, we can if we add a bool flag (var IsInited)... i am not sure we really need this flag to distinguish it, empty Expression / InputString can give the error- it's ok

Reply:
For ex, other Delphi components don't do this additional flag, e.g. Sql connection component just gives an error of connection string is empty

Reply:
wow :( ok thank you for clarification.
of cause if compiler cannot distinguish zero-length and unassingned we have no other choice

Reply:
Unassigned pascal string is a null pointer, when 1+ char is stored to string, it is not null